src/object/srv_obj.c

```diff
@@ -648,10 +776,51 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	if (opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_UPDATE ||
 	    opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_TGT_UPDATE) {
 		bulk_op = CRT_BULK_GET;
-		rc = vos_update_begin(cont->sc_hdl, orw->orw_oid,
-				      orw->orw_epoch, &orw->orw_dkey,
-				      orw->orw_nr, orw->orw_iods.ca_arrays,
-				      &ioh, dth);
+		if (oca->ca_resil == DAOS_RES_EC) {
+			unsigned int tgt_idx = orw->orw_oid.id_shard -
+						orw->orw_start_shard;
+
+			for (i = 0; i < orw->orw_nr; i++)
+				skip_list[i] = NULL;
+			if (tgt_idx >= oca->u.ec.e_p) {
+				rc = ec_data_target(tgt_idx - oca->u.ec.e_p,
+						    orw->orw_nr,
+						    orw->orw_iods.ca_arrays,
+						    oca, skip_list);
+			} else if (tgt_idx == 0) {
+				rc = ec_copy_iods(&iods,
+						  orw->orw_iods.ca_arrays,
+						  orw->orw_nr);
+				if (rc == 0) {
+					rc = ec_parity_target(tgt_idx,
+							      orw->orw_nr,
+							      iods, oca,
+							      skip_list);
+				}
+			} else {
+				rc = ec_parity_target(tgt_idx,
+						      orw->orw_nr,
+						      orw->orw_iods.ca_arrays,
+						      oca, skip_list);
+			}
+			if (rc) {
+				D_ERROR(DF_UOID"EC update failed: %d\n",
+				DP_UOID(orw->orw_oid), rc);
+				goto out;
+			}
+		}
+		if (iods) {
```

Comment:
could add a temporary and set it to either iods or ca_arrays and then have one call here.